sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -86,13 +88,26 @@ export class ApplicationController extends BaseController {
         payload.associatedFiles,
       );
     }
-
+    // TODO:remove the static program year and add dynamic year, from program year table
+    const sequenceName = "2122";
+    let nextApplicationSequence: number = NaN;
+    await this.sequenceService.consumeNextSequence(
+      sequenceName,
+      async (nextSequenceNumber: number) => {
+        nextApplicationSequence = nextSequenceNumber;
+      },
+    );
+    if (!nextApplicationSequence) {
+      throw new BadRequestException();
+    }
+    const applicationNumber =
+      parseInt(sequenceName) * 10 ** 6 + (nextApplicationSequence - 1);
+    submissionResult.data.applicationNumber = applicationNumber;
```

Comment:
The submissionResult.data is supposed to be the result from the application dynamic form data.
Please pass the applicationNumber to the createApplication using a separated variable.